cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -281,3 +289,53 @@ func GetKubeletDir(dryRun bool) (string, error) {
 	}
 	return kubeadmconstants.KubeletRunDirectory, nil
 }
+
+func UpdateKubeletLocalMode(cfg *kubeadmapi.InitConfiguration, dryRun bool) error {
+	// TODO(chrischdi): how to get the correct dir? kubeadm init has a flag to change the location
+	dir := kubeadmconstants.KubernetesDir
+
+	kubeletKubeConfigFilePath := filepath.Join(dir, kubeadmconstants.KubeletKubeConfigFileName)
+
+	if _, err := os.Stat(kubeletKubeConfigFilePath); err != nil {
+		if os.IsNotExist(err) {
+			// TODO(chrischdi): should we print a warning or even return the error?
+			return nil
+		}
+		return err
+	}
+
+	config, err := clientcmd.LoadFromFile(kubeletKubeConfigFilePath)
+	if err != nil {
+		return err
+	}
+
+	configContext, ok := config.Contexts[config.CurrentContext]
+	if !ok {
+		return errors.Errorf("cannot find cluster for active context in kubeconfig %q", kubeletKubeConfigFilePath)
+	}
+
+	localAPIEndpoint, err := kubeadmutil.GetLocalAPIEndpoint(&cfg.LocalAPIEndpoint)
+	if err != nil {
+		return err
+	}
+
+	// Skip changing kubeconfig file if LocalAPIEndpoint is already set.
```

Comment:
better to skip if the Server is anything but the LB / ControlPlaneEndpoint value, because users might have some weird cases.

```
if config.Clusters[configContext.Cluster].Server != cfg.ControlPlaneEndpoint {
	klog.V(2).Infof("Skipping update of the Server value in %s, because the value is already not equal to %s", kubeletKubeConfigFilePath, cfg.ControlPlaneEndpoint)
	return nil
}
```